### ISSUE TITLE
No emoji in bash prompt when launch

### ISSUE THREAD

**Issue by tstringer** [zeit/hyper#645]
I'm seeing an interesting thing.  My bash prompt I have configured an emoji in it, and when launching hyperterm it doesn't recognize it immediately, but it does for all subsequent prompts.  See the screenshot below that shows the repro:

<img width="324" alt="screen shot 2016-08-21 at 6 46 26 pm" src="https://cloud.githubusercontent.com/assets/1445483/17840288/ec98bc08-67cf-11e6-8cd8-64c9fc0964be.png">

**Comment by vwvww** [comment]
I want to work on this issue.
I discovered that:
1. When there is no \n in .bash_profile, this bug does not appear.
![2016-09-10 1 51 37](https://cloud.githubusercontent.com/assets/10724286/18408046/13b16d38-775e-11e6-8d51-a496882e5227.png)
I can reproduce the same bug when there is \n in .bash_profile.
![2016-09-10 1 49 43](https://cloud.githubusercontent.com/assets/10724286/18408058/3e17c7ca-775e-11e6-94b6-16ffab9f1aac.png)

**Comment by Jonas-Drotleff** [comment]
Can you show us your .bash_profile?

**Comment by tstringer** [comment]
@Jonas-Drotleff here it is:

```
export PS1="\u@\h 🌺  \w 
 $ "
```

**Comment by chrishtanaka** [comment]
I have an issue where having the ⚡️(high voltage) emoji always shows ⚡︎ (high voltage sign) instead.
```
# .bash_profile
export PS1='⚡️ \W '
```

**Comment by Jonas-Drotleff** [comment]
@chrishtanaka This is because you use a font that actually has a symbol for the ⚡️ unicode character. The system will always use the font's icons first and if they don't exist, it will take a fallback font, which is the emoji font on macOS

**Comment by chrishtanaka** [comment]
@Jonas-Drotleff How would I determine which fonts have symbols for ⚡️ without trying out each one in hyper?

**Comment by Jonas-Drotleff** [comment]
@chrishtanaka 

This is what I found on the web:
 - [CharMapEx](http://mihai-nita.net/2007/09/08/charmapex-some-kind-of-character-map/)
 - [Character To Glyph Index Mapping Table](https://www.microsoft.com/typography/otspec/cmap.htm)

Basically you just have to try...

**Comment by simeydotme** [comment]
To add to this issue, I find the same thing with any unicode character, even if I try to encode it myself.

```bash
PS1="$ " # no problem
PS1="❭ " # problem
PS1=$'\xe2\x9d\xad\x0a ' # problem
```

The error I see in console is:
![screen shot 2016-12-19 at 11 27 31 am](https://cloud.githubusercontent.com/assets/2817396/21300423/c918d642-c5de-11e6-89b8-064a304398f9.png)

I stepped through the loop, and find that it's running conversion fine on a bunch of characters (although they seem unrelated?) but then it gets to an `undefined` and finally crashes. This method (`lib.UTF8Decoder.prototype.decode`) doesn't seem to run when entering a new line in to the terminal, but only upon opening a new tab.

**Comment by simeydotme** [comment]
Ah, think I've spotted the issue (maybe, I have no idea how to test as the console crashes every time I try to re-compile the code, so I can't patch);

![screen shot 2016-12-19 at 11 40 26 am](https://cloud.githubusercontent.com/assets/2817396/21300565/02628e06-c5e0-11e6-8e1b-013c28b5c1b1.png)

This method here, the loop is using `<=` on the `length` of the string ... (10 for me) but it's zero-indexed and incrementing, meaning the last iteration always has an `undefined` value because it's looking for `length + 1`

**Comment by zpnk** [comment]
Seeing the same issue here. Simple repro:
```bash
# ~/.bash_profile
export PS1='\n 💭 '
```

Results in:

![screen shot 2016-12-21 at 12 13 28 pm](https://cloud.githubusercontent.com/assets/3493970/21404610/e59fec80-c776-11e6-8dcc-6702c5d86375.png)


With the same error as @simeydotme : `Uncaught TypeError: Cannot read property 'length' of undefined`

![screen shot 2016-12-21 at 12 11 59 pm](https://cloud.githubusercontent.com/assets/3493970/21404577/b537ab46-c776-11e6-8ee3-1c96c49e599d.png)

Like @vwvww  mentioned, removing the newline `\n` "fixes" it, but that's obviously not a real solution.

For those bothered by this bug a temporary remedy is to add a `clear` statement the end of your profile which will cause the buggy prompt to be cleared when a new tab/window is opened:

```bash
# ~/.bash_profile
export PS1='\n 💭 '
clear
```

Looks to me though like the `lib.UTF8Decoder.prototype.decode` needs to be fixed to handle `\n`.

**Comment by simeydotme** [comment]
I think this is fixed by #1366 :)
_(at least it's fixed in the latest version; `1.1.0`, for me)_

**Comment by ppot** [comment]
Thanks @simeydotme
